common/src/main/scala/uk/ac/wellcome/message/MessageWorker.scala

```diff
@@ -0,0 +1,62 @@
+package uk.ac.wellcome.message
+
+import akka.actor.ActorSystem
+import uk.ac.wellcome.utils.JsonUtil._
+import uk.ac.wellcome.metrics.MetricsSender
+import uk.ac.wellcome.models.aws.SQSMessage
+import uk.ac.wellcome.utils.JsonUtil.fromJson
+import uk.ac.wellcome.utils.GlobalExecutionContext.context
+
+import scala.concurrent.Future
+import com.twitter.inject.Logging
+
+import scala.concurrent.duration._
+import com.amazonaws.services.s3.AmazonS3
+import io.circe.Decoder
+import uk.ac.wellcome.s3.S3Uri
+import uk.ac.wellcome.s3.S3ObjectStore
+import uk.ac.wellcome.sqs.{SQSReader, SQSWorker}
+import uk.ac.wellcome.utils.JsonUtil._
+
+abstract class MessageWorker[T](sqsReader: SQSReader,
```

Comment:
@alicefuzier would appreciate your input into how to remove the inheritance dependency here.